index.js

```diff
@@ -0,0 +1,311 @@
+        i.classList.add("fas", "fa-ellipsis-v");
+        i.setAttribute("aria-hidden", "true");
+        dropdownButton.appendChild(i);
+        let dropdownMenu = document.createElement('div');
+        dropdownMenu.classList.add("dropdown-menu", "p-2", "flex-column");
+        dropdownMenu.setAttribute("aria-labelledby", "dropdownMenuItem1");
+        dropdownMenu.setAttribute("x-placement", "left-start");
+        dropdownMenu.style = "position: absolute; will-change: transform; top: 0px; left: 0px; transform: translate3d(-162px, 0px, 0px);"
+        dropdown.appendChild(dropdownMenu);
+        let completeButton = document.createElement('button');
+        completeButton.classList.add("btn", "btn-success", "w-100");
+        completeButton.type = "button";
+        completeButton.onclick = moveToComplete;
+        completeButton.innerText = "Complete";
+        dropdownMenu.appendChild(completeButton);
+        let editButton = document.createElement('button');
+        editButton.classList.add("btn", "btn-info", "w-100", "my-2");
+        editButton.type = "button";
+        editButton.setAttribute("data-toggle", "modal");
+        editButton.setAttribute("data-target", "#exampleModal");
+        editButton.onclick = editTaskData;
+        editButton.innerText = "Edit";
+        dropdownMenu.appendChild(editButton);
+        dropdownMenu.appendChild(deleteButton);
+        deleteButton.onclick = deleteTaskData;
+    } else {
+        li.appendChild(deleteButton);
+        deleteButton.style = "width: 100px !important; margin-left: 20px";
+        deleteButton.onclick = deleteTaskData;
+    }
+
+}
+
+function colorTask(li, priority) {
+    switch (priority) {
+        case 'Low priority':
+            li.style.backgroundColor = '#ffc916';
+            break;
+        case 'Medium priority':
+            li.style.backgroundColor = '#a3cd3b';
+            break;
+        case 'High priority':
+            li.style.backgroundColor = '#ff4943';
+            break;
+    }
+}
+
+function createNewTask() {
+    editTaskButton.style.display = "none";
+    addTaskButton.style.display = "inline";
+}
+
+
+function addTask() {
+    const taskCount = taskField.getElementsByTagName("li").length
+    let task = {
+        title: '',
+        text: '',
+        priority: '',
+        time: '',
+        date: ''
+
+    }
+    task.title = title.value;
+    task.text = text.value;
+    task.priority = selectPriority() + ' priority';
+    task.time = setTaskTime();
+    task.date = new Date();
+    toDo.push(task);
+    createTaskField(task, taskField);
+    toDoCount++
+    document.getElementById('toDo-count').innerHTML = ' (' + toDoCount + ')';
+    saveData();
+}
+
+function moveToComplete() {
+    let currentTask = this.closest('.list-group-item');
+    index = Array.from(currentTask.parentNode.children).indexOf(currentTask);
+    taskField.removeChild(currentTask);
+    completed.push(toDo[index]);
+    createTaskField(toDo[index], completeField)

```

Comment:
please follow DRY principle, save code, don't repeat it many times

Reply:
thanks!
fixed it!
```suggestion
let toDoIndex = toDo[index];
```

Reply:
I guess it should be `const todoItem = toDo[index]`. Please, read about difference between let and const I see a lot of errors because you use it wrong. Especially if we speak about primitive and reference types.